### ISSUE TITLE
Address potential JS dependency loading issue for multipage DashR apps

### ISSUE THREAD

**Issue by rpkyle** [plotly/dashR#50]
@chriddyp raised an interesting weakness in the current dependency loading strategy for DashR (via calling the Python-generated JS metadata function (e.g. `.dashHtmlComponents_js_metadata()`).

In a multipage DashR app, it's possible that some dependencies may load properly for the first page, while any new components whose JS dependencies were not initially sourced may fail to load correctly.

There is a clever workaround in Dash for Python, but one does not currently exist in DashR.

I propose that we attempt the following strategy:
- 1. Enumerate the installed packages which have an R package dependency on `dashR`
- 2. Determine which of the dependent packages are currently attached (loaded)
- 3. Check to see if the packages in (2) contain an internal JS metadata function, if so, tack its dependencies on to those already registered by DashR.

Here's an example to demonstrate what I'm proposing (`.packages()` invisibly returns the list, so it's printed here):
```> dependsOnPkgs("ggplot2")
[1] "crosstalk"  "plotly"   
> print(.packages())
[1] "plotly"             "ggplot2"            "tools"              "dashRtutorial"      "magrittr"           "dashCoreComponents" "dashHtmlComponents"
[8] "dashR"              "stats"              "graphics"           "grDevices"          "utils"              "datasets"           "methods"           
[15] "base"  
> deps <- dependsOnPkgs("ggplot2")
> attached <- .packages()
> attached[attached %in% deps]
[1] "plotly"
```

I'm bumping this towards the top of the todo list because it could interfere with DashR docs.

@NadiaTahiri @T4rk1n

**Comment by T4rk1n** [comment]
> 1. Determine which of the dependent packages are currently attached (loaded)

You show me that the other day,  It's a bit like the dash python implementation where importing a library would make it available on the index. :+1:

**Comment by rpkyle** [comment]
Related to a comment @alexcjohnson made a few weeks ago, when the PR for these proposed changes is opened, it will also contain a `add_dependency` method.

`add_dependency` will validate its input, which (for now) will be one or more `htmlDependency` objects, and then attach to the current list of DashR dependencies. A second, private `preload_dependency` method will be written to implement the package inspection logic above.

Since this issue could pose a problem with documentation serving, and should not take long to address, I'm planning to address it once I've addressed review comments in #53.

@chriddyp

**Comment by rpkyle** [comment]
Following a discussion with @alexcjohnson a while ago, I've opened a PR (#97) which proposes that we retrieve metadata from attached packages rather than via introspection of the initial Dash layout.

This should resolve the multipage issue; the code seems slightly cleaner too.

**Comment by rpkyle** [comment]
Closed by #97.
